### PR TITLE
Create ts-source-parser package

### PR DESCRIPTION
**Pull request by jean-michelet**
This PR follows on from discussion [#214](https://github.com/NodeSecure/js-x-ray/issues/214), which proposes a new package in workspaces for parsing TypeScript files. 

## Usage example
```js
import { TsSourceParser } from "@nodesecure/ts-source-parser";

const parser = new TsSourceParser();
const body = parser.parse("const x: number = 5;");

console.log(body);
```

## Usage with `js-x-ray` 
```js
import { runASTAnalysis } from "@nodesecure/js-x-ray";
import { readFileSync } from "node:fs";

const { warnings, dependencies } = runASTAnalysis(
    readFileSync("./file.ts", "utf-8"),
    { customParser: new TsSourceParser() }
);

console.log(dependencies);
console.dir(warnings, { depth: null });
```

### CODE REVIEW

File: .github/workflows/ts-source-parser.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
```
Comment (PierreDemailly) [inline .github/workflows/ts-source-parser.yml]:
```suggestion
name: TsSourceParser CI
```

The idea is to have a different name for this workspace so we can identify it in runned jobs

File: .github/workflows/ts-source-parser.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      with:
+        egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm install
+    - run: npm test
```
Comment (PierreDemailly) [inline .github/workflows/ts-source-parser.yml]:
```suggestion
    - run: npm run test --workspace=workspaces/ts-source-parser
```

File: workspaces/ts-source-parser/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+    "name": "@nodesecure/ts-source-parser",
```
Comment (PierreDemailly) [inline workspaces/ts-source-parser/package.json]:
```suggestion
  "name": "@nodesecure/ts-source-parser",
```
We indents with 2 spaces (need to update the whole file). If you use VSC do you have EditorConfig extension enabled ?

**Review by PierreDemailly — COMMENTED** [review]
You also need to add the workspace in the root package.json
https://github.com/NodeSecure/js-x-ray/blob/f4b47c72b7996261df264f91757c33bfcac97406/package.json#L21

Everything else lgtm

File: .github/workflows/ts-source-parser.yml (hunk 2 above)
Comment (jean-michelet) [reply .github/workflows/ts-source-parser.yml]: Ok, but this flag does not exist in other workspaces, do you want to add them as well?

Comment (PierreDemailly) [reply .github/workflows/ts-source-parser.yml]: Well, yes you can ^^

**Review by PierreDemailly — APPROVED** [review]

**Pull request merged**